pyht/async_client.py

```diff
@@ -131,12 +149,28 @@ async def tts(
         text = ensure_sentence_end(text)
 
         request = api_pb2.TtsRequest(params=options.tts_params(text, voice_engine), lease=lease_data)
-        stub = api_pb2_grpc.TtsStub(self._rpc[1])
-        stream: TtsUnaryStream = stub.Tts(request)
-        if context is not None:
-            context.assign(stream)
-        async for response in stream:
-            yield response.data
+        try:
+            stub = api_pb2_grpc.TtsStub(self._rpc[1])
+            stream: TtsUnaryStream = stub.Tts(request)
+            if context is not None:
+                context.assign(stream)
+            async for response in stream:
+                yield response.data
+        except grpc.RpcError as e:
+            error_code = getattr(e, "code")()
+            should_fallback = (
+                    error_code is grpc.StatusCode.RESOURCE_EXHAUSTED
+                    or error_code is grpc.StatusCode.UNAVAILABLE
+            )
```

Comment:
nit: I despise nesting :grin: 
It's a little cleaner to raise out of this here like:
```
if error_code not in {RESOURCE_EXHAUSTED, UNAVAILABLE} or self._fallback_rpc is None:
    raise
# Continue with rest of code, no need to check `should_fallback`
```

Reply:
Fixed.